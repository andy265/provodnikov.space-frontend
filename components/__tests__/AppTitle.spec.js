const { createRenderer } = require('vue-server-renderer')
const { shallow } = require('vue-test-utils')
const AppTitle = require('~/components/AppTitle')

describe('AppTitle.vue', () => {
  let tags = ['h1', 'h2', 'h3', 'span']
  let themes = ['text', 'sub', 'normal', 'main', '404']

  themes.forEach(theme => {
    it(`has same HTML structure with theme ${theme}`, () => {
      let wrapper = shallow(AppTitle, {
        propsData: {
          tag: 'span',
          theme: theme
        },
        slots: {
          default: '<span>some text</span>'
        }
      })
      let renderer = createRenderer()
      // eslint-disable-next-line
      renderer.renderToString(wrapper.vm, (err, str) => {
        expect(str).toMatchSnapshot()
      })
    })
  })

  it('can have link', () => {
    let wrapper = shallow(AppTitle, {
      propsData: {
        tag: 'span',
        to: '/page',
        theme: 'text'
      }
    })
    expect(wrapper.find('a').hasAttribute('href', '/page')).toBeTruthy()
  })

  tags.forEach(tag => {
    it(`can have tag ${tag}`, () => {
      let wrapper = shallow(AppTitle, {
        propsData: {
          tag: tag,
          theme: 'text'
        }
      })
      expect(wrapper.find(tag).element).toBeTruthy()
    })
  })
})
