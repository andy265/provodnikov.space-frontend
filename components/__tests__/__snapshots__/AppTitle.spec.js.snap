// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppTitle.vue has same HTML structure with theme 404 1`] = `
<div class="app-title app-title_theme_404"><span class="app-title__tag"><span>some text</span></span>
</div>
`;

exports[`AppTitle.vue has same HTML structure with theme main 1`] = `
<div class="app-title app-title_theme_main"><span class="app-title__tag"><span>some text</span></span>
</div>
`;

exports[`AppTitle.vue has same HTML structure with theme normal 1`] = `
<div class="app-title app-title_theme_normal"><span class="app-title__tag"><span>some text</span></span>
</div>
`;

exports[`AppTitle.vue has same HTML structure with theme sub 1`] = `
<div class="app-title app-title_theme_sub"><span class="app-title__tag"><span>some text</span></span>
</div>
`;

exports[`AppTitle.vue has same HTML structure with theme text 1`] = `
<div class="app-title app-title_theme_text"><span class="app-title__tag"><span>some text</span></span>
</div>
`;
